crates/router/src/core/unified_connector_service.rs

```diff
@@ -744,3 +745,123 @@ pub fn extract_webhook_content_from_ucs_response(
 ) -> Option<&unified_connector_service_client::payments::WebhookResponseContent> {
     transform_data.webhook_content.as_ref()
 }
+
+/// UCS Event Logging Wrapper Function
+/// This function wraps UCS calls with comprehensive event logging.
+/// It logs the actual gRPC request/response data, timing, and error information.
+#[instrument(skip_all, fields(connector_name, flow_type, payment_id))]
+pub async fn ucs_logging_wrapper<T, F, Fut, Req, Resp, GrpcReq, GrpcResp>(
+    router_data: RouterData<T, Req, Resp>,
```

Comment:
I think This can be a mutable ref. Can avoid cloning router data as well
```suggestion
    router_data: &mut RouterData<T, Req, Resp>,
```

Reply:
The mutable reference approach creates lifetime issues with async closures - Rust can't move borrowed references in `async move` blocks.